crypto/src/main/java/net/corda/v5/crypto/DigestAlgorithmName.java

```diff
@@ -0,0 +1,83 @@
+package net.corda.v5.crypto;
+
+import java.util.Locale;
+import net.corda.v5.base.annotations.CordaSerializable;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+/**
+ * The digest algorithm name. This class is to be used in Corda hashing API.
+ *
+ * @param name The name of the digest algorithm to be used for the instance.
+ */
+
+@CordaSerializable
+public final class DigestAlgorithmName {
+    @NotNull private final String name;
+    /**
+     * Instance of SHA-256
+     */
+    @NotNull public static final DigestAlgorithmName SHA2_256 = new DigestAlgorithmName("SHA-256");
```

Comment:
`NotNull` seems overkill to me here I think. Same for below cases.

Reply:
It's correct, though, and better to be consistent.